cli/medperf/entities/report.py

```diff
@@ -0,0 +1,138 @@
+import hashlib
+import os
+import yaml
+import logging
+from typing import List, Union, Optional
+
+from medperf.utils import storage_path
+from medperf.entities.schemas import MedperfBaseSchema
+import medperf.config as config
+from medperf.exceptions import InvalidArgumentError
+
+
+class TestReport(MedperfBaseSchema):
```

Comment:
I see. I strongly believe that there is no need for branching for the entity classes like we have for the schemas, especially because we have only 5 implementations, not that much. I wonder how neat that will appear. If we want to inherit from Entity here I would suggest then simply not implementing the unsupported methods, or implement them with raising some nonimplementederror or something similar.

Reply:
I think I was wrong. It seems best practice not to overwrite the upload method as I suggested, and a mixin will be the only option :)